zebra-rpc/src/methods.rs

```diff
@@ -2078,7 +2073,22 @@ where
             last_output_location = output_location;
         }
 
-        Ok(response_utxos)
+        if !utxos_request.chain_info {
+            Ok(GetAddressUtxosResponse::ChainInfoFalse(response_utxos))
+        } else {
+            let (height, hash) = self
+                .latest_chain_tip
+                .best_tip_height_and_hash()
+                .ok_or_misc_error("No blocks in state")?;
```

Comment:
Could we update the `UtxosByAddresses` `ReadRequest` to return the chain tip height/hash instead of reading it here?

I'm not sure how clients will be using the requested chain info, but it's possible that the best chain tip could change after the state handles the `UtxosByAddresses` request so that it looks like the returned utxos are for a later chain tip than what was queried.

Reply:
Documented in 5152fbead, I also simplified the state response by incorporating the height/hash into `AddressUtxos`

Reply:
This turned out to be more complicated than I would've thought, thank you for fixing it!